Update scaladoc docs (#12787)
lampepfl/dotty: docs/docs/usage/scaladoc/settings.md

##### -siteroot

A directory containing static files from which to generate documentation. Default directory is `./docs`

> ##### -versions-dictionary-url

Reviewer: It would be nice to show an example of sbt expression setting the option.

Author: I can add this one, but in my opinion it's a basic knowledge how to add setting to sbt.

Reviewer: I agree, but I think it’s often good to give copy-pastable examples.

> A URL pointing to a JSON document containing a dictionary version -> documentation location.

Reviewer 2: ```suggestion
A URL pointing to a JSON document containing a dictionary: `version label -> documentation location`.
```

> The JSON file has single property "versions" that holds dictionary of labels of specific docs and URL pointing to their index.html top-level file.

Reviewer 2: ```suggestion
The JSON file has single property `versions` that holds the dictionary associating the labels of specific versions of the documentation to the URLs pointing to their index.html
```

Its argument must be a URL to a JSON document holding information about the locations of specific versions.

> Useful for libraries that maintain different releases docs.

Reviewer 2: ```suggestion
Useful for libraries that maintain different versions of their documentation
```

Example JSON file:
```
{
  "versions": {

>     "3.0.X": "https://dotty.epfl.ch/3.0.X/docs/index.html",

Reviewer: ```suggestion
    "3.0.x": "https://dotty.epfl.ch/3.0.x/docs/index.html",
```
